exporter/elasticsearchexporter/model.go

```diff
@@ -244,23 +361,52 @@ func (m *encodeModel) encodeSpanDefaultMode(resource pcommon.Resource, span ptra
 	document.AddInt("TraceStatus", int64(span.Status().Code()))
 	document.AddString("TraceStatusDescription", span.Status().Message())
 	document.AddString("Link", spanLinksToString(span.Links()))
-	m.encodeAttributes(&document, span.Attributes(), idx)
-	document.AddAttributes("Resource", resource.Attributes())
-	m.encodeEvents(&document, span.Events())
+	document.AddAttributes("Resource", ec.resource.Attributes())
 	document.AddInt("Duration", durationAsMicroseconds(span.StartTimestamp().AsTime(), span.EndTimestamp().AsTime())) // unit is microseconds
-	document.AddAttributes("Scope", scopeToAttributes(scope))
-	return document
+	document.AddAttributes("Scope", scopeToAttributes(ec.scope))
+	encodeAttributes(e.attributesPrefix, &document, span.Attributes(), idx)
+	document.AddEvents(e.eventsPrefix, span.Events())
+	return document.Serialize(buf, e.dedot)
 }
 
-func (m *encodeModel) encodeSpanEvent(resource pcommon.Resource, resourceSchemaURL string, span ptrace.Span, spanEvent ptrace.SpanEvent, scope pcommon.InstrumentationScope, scopeSchemaURL string, idx elasticsearch.Index, buf *bytes.Buffer) {
-	if m.mode != MappingOTel {
-		// Currently span events are stored separately only in OTel mapping mode.
-		// In other modes, they are stored within the span document.
-		return
+type ecsDataPointsEncoder struct{}
+
+func (ecsDataPointsEncoder) encodeMetrics(
+	ec encodingContext,
+	dataPoints []datapoints.DataPoint,
+	validationErrors *[]error,
+	idx elasticsearch.Index,
+	buf *bytes.Buffer,
+) (map[string]string, error) {
+	dp0 := dataPoints[0]
+	var document objmodel.Document
+	encodeAttributesECSMode(&document, ec.resource.Attributes(), resourceAttrsConversionMap, resourceAttrsToPreserve)
+	document.AddTimestamp("@timestamp", dp0.Timestamp())
+	document.AddAttributes("", dp0.Attributes())
+	addDataStreamAttributes(&document, "", idx)
+
+	for _, dp := range dataPoints {
+		value, err := dp.Value()
+		if err != nil {
+			*validationErrors = append(*validationErrors, err)
+			continue
+		}
+		document.AddAttribute(dp.Metric().Name(), value)
+	}
+	err := document.Serialize(buf, true)
+
+	return document.DynamicTemplates(), err
+}
+
+func addDataStreamAttributes(document *objmodel.Document, key string, idx elasticsearch.Index) {
+	if idx.IsDataStream() {
+		document.AddString(key+"data_stream.type", idx.Type)
+		document.AddString(key+"data_stream.dataset", idx.Dataset)
+		document.AddString(key+"data_stream.namespace", idx.Namespace)
 	}
-	otelserializer.SerializeSpanEvent(resource, resourceSchemaURL, scope, scopeSchemaURL, span, spanEvent, idx, buf)
 }
 
+/*
```

Comment:
is it intentional to keep this as a comment?

Reply:
Nope, thanks/sorry